lib/remote_node.go

```diff
@@ -0,0 +1,468 @@
+package lib
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/deso-protocol/core/bls"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/sha3"
+	"net"
+	"time"
+)
+
+type RemoteNodeConnectionType int
+
+const (
+	RemoteNodeConnectionType_Unknown  RemoteNodeConnectionType = 0
+	RemoteNodeConnectionType_Outbound RemoteNodeConnectionType = 1
+	RemoteNodeConnectionType_Inboound RemoteNodeConnectionType = 2
+)
+
+type RemoteNodeConnectionStatus int
+
+const (
+	RemoteNodeConnectionStatus_NotConnected RemoteNodeConnectionStatus = 0
+	RemoteNodeConnectionStatus_Connected    RemoteNodeConnectionStatus = 1
+	RemoteNodeConnectionStatus_Attempted    RemoteNodeConnectionStatus = 2
+	RemoteNodeConnectionStatus_Terminated   RemoteNodeConnectionStatus = 3
+)
+
+type RemoteNode struct {
+	// Should we have ID here?
+	peer             *Peer
+	id               RemoteNodeId
+	connectionType   RemoteNodeConnectionType
+	connectionStatus RemoteNodeConnectionStatus
+
+	params *DeSoParams
+
+	srv  *Server
+	bc   *Blockchain
+	cmgr *ConnectionManager
+
+	minTxFeeRateNanosPerKB uint64
+	hyperSync              bool
```

Comment:
Do we also want to know if it's archival or not?

Reply:
We indeed do lol.